video/out/wayland_common.c

```diff
@@ -2020,9 +2020,11 @@ bool vo_wayland_init(struct vo *vo)
         wl->video_viewport = wp_viewporter_get_viewport(wl->viewporter, wl->video_surface);
     }
 
-    const char *xdg_current_desktop = getenv("XDG_CURRENT_DESKTOP");
-    if (xdg_current_desktop != NULL && strstr(xdg_current_desktop, "GNOME"))
-        MP_WARN(wl, "GNOME's wayland compositor lacks support for the idle inhibit protocol. This means the screen can blank during playback.\n");
+    if (!wl->idle_inhibit_manager) {
+        const char *xdg_current_desktop = getenv("XDG_CURRENT_DESKTOP");
+        if (xdg_current_desktop != NULL && strstr(xdg_current_desktop, "GNOME"))
+            MP_WARN(wl, "GNOME's wayland compositor lacks support for the idle inhibit protocol. This means the screen can blank during playback.\n");
```

Comment:
You could move this to the already existing `if (!wl->idle_inhibit_manager)` block further below.